modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/ApiTransporter.swift

```diff
@@ -15,19 +15,29 @@ public protocol ApiTransporter {
 }
 
 final class ApiTransporterRetryPolicy: RetryPolicy {
+    private let timeout: TimeInterval
 
-    init() {
+    init(timeout: TimeInterval) {
+        self.timeout = timeout
         super.init(retryableHTTPStatusCodes: Set([429, 408, 500, 502, 503, 504]))
     }
 
     override func retry(_ request: Alamofire.Request,
                         for session: Session,
                         dueTo error: Error,
                         completion: @escaping (RetryResult) -> Void) {
+        let d = request.cURLDescription()
+        if d.contains("api.coingeck") {
+            print("xxx.retry for \(request.cURLDescription()) error: \(error)")
+        }
 
         if request.retryCount < retryLimit, shouldRetry(request: request, dueTo: error) {
             if let httpResponse = request.response, let delay = ApiTransporterRetryPolicy.retryDelay(from: httpResponse) {
-                completion(.retryWithDelay(delay))
+                if delay <= timeout {
```

Comment:
Based on:

```
if delay <= timeout {
    completion(.retryWithDelay(delay))
} else {
    completion(.doNotRetry)
}
```

and:

> * Timeout — we send a request and start waiting for a response. The max time we wait before dropping it is the timeout. This might be say, 1 to 10 seconds
> * Time to retry returned by CoinGecko — this is the time that CoinGecko says we should hold off before making requests again. In other words, we are rate limited. This could be say, 600 seconds

`delay` is "time to retry" right?

So:

```
if delay_600 <= timeout_60
    retry
else
    do_not_retry
```

I don't think they should be compared. They are different concepts.

We set a timeout of `60`, that's fine. That's because we presume the API takes a while to respond. Then it should never change.

And if the API responds that we are rate limited, that we should back off for at least 600s, then we should do something like this:

```
func call_coingecko()
    .request(timeout_60)
    .process {
        if delay
            sleep(delay)
            call_coingecko() #retry, only a number of times
        else
            return results
        end
    }
```

Reply:
ok got it, than leave as it is for now